src/ert/shared/feature_toggling.py

```diff
@@ -61,6 +63,18 @@ def update_from_args(args: Namespace) -> None:
                 logger = logging.getLogger()
                 logger.warning(FeatureToggling._conf[feature_name].msg)
 
+    @staticmethod
+    def update_from_environment() -> None:
+        pattern = re.compile("(?:ERT_)(ENABLE_|DISABLE_)(.*)")
+        args = dict()
+        for key, value in os.environ.items():
+            if pattern.match(key):
+                feature_name = key[4:]
+                feature_name_arg = feature_name.replace("-", "_").lower()
+                args[feature_name_arg] = value
```

Comment:
Your order of operations in the `str.replace` function is wrong (it needs to replace `_` with `-`). Please add a test for this.

Also, I reckon explicit disable doesn't make sense in environment variables, where an empty string and `0` already mean disable? Like, what is the correct behaviour of `ERT_ENABLE_SCHEDULER=1 ERT_DISABLE_SCHEDULER=1`, or `ERT_DISABLE_SCHEDULER= ert test_run`?

```suggestion
        pattern = re.compile("^ERT_(ENABLE_|DISABLE_)(.+)$")
        args: MutableMapping[str, str] = {}
        for key, value in os.environ.items():
            if (match := pattern.match(key)):
                feature = match[2].replace("_", "-").lower()
                args[feature] = match[1] == "ENABLE_"
```

Reply:
it is done that way on purpose to align with how update_from_args works. this way I could reuse all of update_from_args without making changes to it.

Reply:
code was rewritten. this is not relevant anymore